e2e/k8s/vault-mtls.values.yaml

```diff
@@ -0,0 +1,30 @@
+injector:
+  enabled: false
+server:
+  extraEnvironmentVars:
+    VAULT_CACERT: /etc/vault-config/vault-server-ca.pem
+    VAULT_ADDR: https://127.0.0.1:8200
+    VAULT_CLIENT_CERT: /etc/vault-config/es-client.pem
+    VAULT_CLIENT_KEY: /etc/vault-config/es-client-key.pem
+  volumeMounts:
+    - name: tls-config
+      mountPath: /etc/vault-config
+      readOnly: true
+  volumes:
+    - name: tls-config
+      secret:
+        secretName: vault-tls-config
+  standalone:
+    config: |
+      ui = true
+      listener "tcp" {
+        address = "[::]:8200"
+        cluster_address = "[::]:8201"
+        tls_cert_file = "/etc/vault-config/server-cert.pem"
+        tls_key_file = "/etc/vault-config/server-cert-key.pem"
+        tls_client_ca_file = "/etc/vault-config/vault-client-ca.pem"
+        tls_require_and_verify_client_cert = true
+      }
```

Comment:
From what i can see the only difference is `tls_require_and_verify_client_cert = true`, would you think it would be possible to add a second `listener "tcp" {}` section to the existing `vault.values.yaml` file ? 
Then we can safe a couple of if/else branches in the e2e test code. Then we can just create another store which points to a different vault port - the one that requires mTLS.

Reply:
@moolen , I tried to added the new listener, while that work fine from the perspective of the helm install, the Vault chart does not allow customising the Service ports [here](https://github.com/hashicorp/vault-helm/blob/v0.11.0/templates/server-service.yaml#L27), which would make it difficult to get a valid address to point the clients to the mTLS endpoint, I thought about creating a new Service manually for the mTLS port, however that doesn't seem an elegant solution, thus I suggest keeping my original implementation, which would perform the Vault installation with a single listener having mTLS configured under it.

Reply:
> Vault chart does not allow customising the Service ports

oh no, that's a pity :(

I think having a secondary service is probably the most elegant solution we can get :D 
In the vault addon there is a controller-runtime client which can be used to create a service in the relevant namespace. I strong-ishly prefer this over having a bunch of if/else branches in a bunch of files to make it work.

Reply:
No problem @moolen , I have amend my PR and switched back to a single values file with the additional listener, instead of creating a new Vault Service, I decided to patch the existing one and add the extra port to it. Could you please have another look if the simplifications provided in the e2e tests are now good enough? Thanks in advance.